frontend/public/components/catalog.jsx

```diff
@@ -0,0 +1,188 @@
+        obj: serviceClass,
+        kind,
+        tileName,
+        tileIconClass,
+        tileImgUrl,
+        tileDescription,
+        tileProvider,
+        callback: () => {
+          history.push(`/k8s/cluster/clusterserviceclasses/${serviceClass.metadata.name}/create-instance`);
+        }
+      };
+    });
+  }
+
+  normalizeImagestreams(imagestreams, kind) {
+    imagestreams = _.map(imagestreams, imagestream => {
+      if (isBuilder(imagestream)) {
+        const tag = getMostRecentBuilderTag(imagestream);
+        const tileName = _.get(imagestream, ['metadata', 'annotations', 'openshift.io/display-name']) || imagestream.metadata.name;
+        const iconClass = getImageStreamIcon(tag);
+        const tileImgUrl = getImageForIconClass(iconClass);
+        const tileIconClass = tileImgUrl ? null : iconClass;
+        const tileDescription = _.get(tag, 'annotations.description');
+        const tileProvider = _.get(tag, 'annotations.openshift.io/provider-display-name');
+        const { name, namespace } = imagestream.metadata;
+        return {
+          obj: imagestream,
+          kind,
+          tileName,
+          tileIconClass,
+          tileImgUrl,
+          tileDescription,
+          tileProvider,
+          callback: () => {
+            history.push(`/source-to-image?imagestream=${name}&ns=${namespace}`);
+          }
+        };
+      }
+      return {
+        hidden: true
+      };
+    });
+
+    return _.filter(imagestreams, imagestream => {
+      return !imagestream.hidden;
+    });
+  }
+
+  render() {
+    const {loaded, loadError} = this.props;
+    const {items} = this.state;
+
+    return <div className="co-m-pane">
+      <div className="co-m-pane__body">
+        <StatusBox data={items} loaded={loaded} loadError={loadError} label="Resources">
+          <CatalogList items={items} />
+        </StatusBox>
+      </div>
+    </div>;
+  }
+}
+
+CatalogListPage.displayName = 'CatalogList';
+
+CatalogListPage.propTypes = {
+  obj: PropTypes.object
+};
+
+// eventually may use namespace
+// eslint-disable-next-line no-unused-vars
+export const Catalog = connectToFlags(FLAGS.OPENSHIFT, FLAGS.SERVICE_CATALOG)(({namespace, flags}) => {
+
+  if (flagPending(flags.OPENSHIFT) || flagPending(flags.SERVICE_CATALOG)) {
+    return null;
+  }
+
+  const resources = [];
+  if (flags.SERVICE_CATALOG) {
+    resources.push({
+      isList: true,
+      kind: 'ClusterServiceClass',
+      namespaced: false,
+      prop: 'clusterserviceclasses'
+    });
+  }
+  if (flags.OPENSHIFT) {
+    resources.push({
+      isList: true,
+      kind: 'ImageStream',
+      namespace: 'openshift',
+      prop: 'imagestreams'
+    });
+  }
+  return <div className="catalog">
+    <Firehose resources={resources}>
+      <CatalogListPage />
+    </Firehose>
+  </div>;
+});
+
+Catalog.displayName = 'Catalog';
+
+Catalog.propTypes = {
+  namespace: PropTypes.string.isRequired,
+};
+
+
+class CatalogPage_ extends React.PureComponent {
```

Comment:
This should be a stateless functional component.

```js
export const CatalogPage = ({match}) => {
  const namespace = _.get(match, 'params.ns');
  return <React.Fragment>
    <Helmet>
      <title>Catalog</title>
    </Helmet>
    <NavTitle title="Catalog" />
     <Catalog namespace={namespace} />
  </React.Fragment>;
};

Reply:
fixed